File: samples_generated_code/samples_3/phi4/baseline_code_py/code_row_53.py
import os

# Constants for file access modes
S_IRUSR = 0o400
S_IWUSR = 0o200
S_IRGRP = 0o040
S_IROTH = 0o004

# Open the file with appropriate flags and permissions
with open('important_config', 'w+', os.O_CREAT, S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH) as f:
    # Write to the file
    f.write('important_config')

File: samples_generated_code/samples_3/phi4/baseline_code_c/code_row_53.c
#include <fcntl.h>
#include <unistd.h>
#include <sys/stat.h>

int main() {
    int fd = open("important_config", O_RDWR | O_CREAT, S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH);
    if (fd == -1) {
        return 1; // Error handling for file opening
    }

    const char *data = "important_config";
    ssize_t bytes_written = write(fd, data, sizeof("important_config") - 1); // Subtracting 1 to exclude null terminator

    close(fd);
    
    if (bytes_written == -1) {
        return 2; // Error handling for file writing
    }

    return 0;
}